addons/sale/models/sale_order.py

```diff
@@ -1152,6 +1152,11 @@ def _prepare_invoice(self):
         """
         self.ensure_one()
 
+        txs_to_be_linked = self.transaction_ids.filtered(
+            lambda tx: (tx.state == 'done' and not (tx.payment_id and tx.payment_id.is_reconciled))
+                       or tx.state in ('pending', 'authorized')
```

Comment:
```suggestion
            lambda tx: (
                tx.state in ('pending', 'authorized')
                or tx.state == 'done'
                and not (tx.payment_id and tx.payment_id.is_reconciled)
            )
```
feels clearer to me, but feel free to find another clean way to display it (I feel like the display is not really clear rn)

Reply:
I understand why you think so, but I don't like getting rid of the parenthesis like that, even if correct, for clarity purpose too. Would this be okay for you? As they are still on the same line, even without parenthesis, it's still clear it's the same condition
```suggestion
            lambda tx: (
                tx.state in ('pending', 'authorized')
                or tx.state == 'done' and not (tx.payment_id and tx.payment_id.is_reconciled)
            )
```